numpy/_core/tests/_natype.py

```diff
@@ -14,10 +14,8 @@ def _create_binary_propagating_op(name, is_divmod=False):
     def method(self, other):
         if (
             other is pd_NA
-            or isinstance(other, (str, bytes))
-            or isinstance(other, (numbers.Number, np.bool))
-            or isinstance(other, np.ndarray)
-            and not other.shape
+            or isinstance(other, (str, bytes, numbers.Number, np.bool))
+            or isinstance(other, np.ndarray) and not other.shape
```

Comment:
I think the previous was more readable.

Reply:
Not sure about it. Calling the `isinstance` function twice doesn't seem more readable to me.

Reply:
Would we apply these automatically (Maybe even as a `pre-commit`) in a future?
In that case, I happy to just do it.  Otherwise it is matter of opinion (the old code grouped things nicely).

If we enforce/use more auto-formatting, I suspect that one add comments to avoid it if desired (I don't think this instance matters much):
```
        or isinstance(other, (str, bytes))  # str-like
```
which is also true for some of the other simplification rules.

Reply:
I would be for applying some micro-optimisations automatically using the ruff linter. They would still be micro-optimisations of course, but I think that's OK if they're applied in a systematic and automatic way.

This would be such a micro-optimisation applied by the ruff linter: one single `isinstance` call instead of multiple ones - more than a formatting issue applied by the ruff formatter.